docs/v2/guides/performance.md

```diff
@@ -158,6 +105,69 @@ $loader = new Timber\Loader();
 $loader->clear_cache_twig();
 ```
 
+
+## Cache _Parts_ of the Twig File and Data
+
+If you want to use [`cache` tag](https://twig.symfony.com/doc/3.x/tags/cache.html) in Twig, you’ll have to install the `twig/cache-extra` package:
+
+```bash
+composer require twig/cache-extra
+```
+
+And register it in Timber like this:
+
+**functions.php**
+
+```php
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Component\Cache\Adapter\TagAwareAdapter;
+use Twig\Extra\Cache\CacheExtension;
+use Twig\Extra\Cache\CacheRuntime;
+use Twig\RuntimeLoader\RuntimeLoaderInterface;
+
+add_filter('timber/twig', function ($twig) {
+		$twig->addRuntimeLoader(new class implements RuntimeLoaderInterface
+		{
+			public function load($class)
+			{
+				if (CacheRuntime::class === $class) {
+					return new CacheRuntime(new TagAwareAdapter(new FilesystemAdapter('', 0, TIMBER_LOC . '/cache/twig')));
+				}
+			}
+		});
+		$twig->addExtension(new CacheExtension());
+
+    return $twig;
+});
+```
+
+You can then use it like this:
+
+```twig
+{% cache 'index;content' %}
+    {% for post in posts %}
+        {% include ['tease-' ~ post.post_type ~ '.twig', 'tease.twig'] %}
+    {% endfor %}
+{% endcache %}
+```
+
+Read more about it in [Twig’s `cache` documentation](https://twig.symfony.com/doc/3.x/tags/cache.html).
+
+If you want to use something meaningful for the cache key, you can also generate a cache key from the variables that you use. In the following example, we generate a cache key from `$posts` and then generate a key from it:
+
+```php
+$generator = new Timber\Cache\KeyGenerator();
+$key = $generator->generateKey($posts);
+```
+
+### Extra: TimberKeyGeneratorInterface
+
+Instead of hashing a whole object, you can specify the cache key in the object itself. If the object implements `TimberKeyGeneratorInterface`, it can pass a unique key through the method `get_cache_key`. That way a class can for example simply pass `'last_updated'` as the unique key.
```

Comment:
```suggestion
Instead of hashing a whole object, you can specify the cache key in the object itself. If the object implements `TimberKeyGeneratorInterface`, it can pass a unique key through the method `get_cache_key()`. That way a class could for example pass a `'last_updated'` property as the unique key.
```